src/pages/Hojung/Main/feeds/CommentList.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+
+class CommentList extends React.Component {
+  commentLike = e => {
+    const { commentData, commentLikeState } = this.props;
+    const { parentElement } = e.target;
+    const _comments = commentData;
+    _comments.map(el => {
+      if (el.id === Number(parentElement.id)) {
+        el.isLiked = !el.isLiked;
+      }
```

Comment:
map 메서드가 새로운 배열을 리턴하기는 하지만 if 문 내부에 있는 로직들은 새로운 값을 만들어내는게 아니라 기존 값을 수정하는 방식으로 구현되고 있네요. 얕은 복사 / 깊은 복사, 참조형 데이터의 불변성에 대해 검색해보시고 이해하신 부분을 기반으로 어떻게 고쳐볼 수 있을지 댓글로 남겨주세요!

Reply:
종택님 댓글 확인 후 , 몇시간동안 공부했지만 쉽지 않은 개념이네요 🥺
주말동안 완벽히 이해한 뒤 다시 댓글 달겠습니다!

Reply:
[얕은 복사 / 깊은 복사 , 참조형 데이터 공부하여 블로깅 했습니다!](https://velog.io/@jjung-developer/TIL-31-%EC%96%95%EC%9D%80-%EB%B3%B5%EC%82%ACshallow-copy-VS-%EA%B9%8A%EC%9D%80-%EB%B3%B5%EC%82%ACdeep-copy)

제일 처음 _comments를 commentData에서 복사 해 올때 Object.assign({}, commentData); 를 사용
혹은
```
_comments.map(el => {
      const obj = Object.assign({}, el);
      if (el.id === Number(parentElement.id)) {
        obj.isLiked = !obj.isLiked;
      }
```

이런 식으로 깊은 복사를 할 수 있을것 같습니다!

Reply:
```suggestion
    _comments.map(comment => {
      if (comment.id === +parentElement.id) {
        return { ...comment, isLiked: !comment.isLiked };
      } else {
        return comment
      }
```
위와 같이 적어줄 수도 있을 것 같습니다. 그런데 로직을 적다보니 몇 가지 의문이 드는데
- id가 일치하는 경우가 아닌 경우에는 undefined를 리턴하는 것이 맞나요?
- 1) 맞다면 map 대신 filter를 사용해야 할거 같고
- 2) 아니라면 else인 케이스에 대해서 comment를 리턴해줘야 원래 정보도 보존한 채로 새로운 배열이 리턴될거 같습니다